File: src/nl/rubensten/texifyidea/modules/LatexModuleType.java
package nl.rubensten.texifyidea.modules;

import com.intellij.openapi.module.ModuleType;
import com.intellij.openapi.module.ModuleTypeManager;
import nl.rubensten.texifyidea.TexifyIcons;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * @author Sten Wessel
 */
public class LatexModuleType extends ModuleType<LatexModuleBuilder> {

    private static final String ID = "LATEX_MODULE_TYPE";

    public LatexModuleType() {
        super(ID);
    }

    public static LatexModuleType getInstance() {
        return (LatexModuleType)ModuleTypeManager.getInstance().findByID(ID);
    }

    @NotNull
    @Override
    public LatexModuleBuilder createModuleBuilder() {
        return new LatexModuleBuilder();
    }

    @NotNull
    @Override
    public String getName() {
        return "LaTeX";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "LaTeX";
    }

    @Override
    public Icon getBigIcon() {
        return TexifyIcons.LATEX_MODULE;
    }

    @Override
    public Icon getNodeIcon(@Deprecated boolean isOpened) {
        return TexifyIcons.LATEX_MODULE;
    }
}

File: src/nl/rubensten/texifyidea/modules/LatexModuleType.kt
package nl.rubensten.texifyidea.modules

import com.intellij.openapi.module.ModuleType
import com.intellij.openapi.module.ModuleTypeManager
import nl.rubensten.texifyidea.TexifyIcons

/**
 * @author Sten Wessel
 */
class LatexModuleType : ModuleType<LatexModuleBuilder>(ID) {

    companion object {

        private const val ID = "LATEX_MODULE_TYPE"

        val INSTANCE: LatexModuleType
            get() = ModuleTypeManager.getInstance().findByID(ID) as LatexModuleType
    }

    override fun createModuleBuilder() = LatexModuleBuilder()

    override fun getName() = "LaTeX"

    override fun getDescription() = "LaTeX"

    override fun getNodeIcon(isOpened: Boolean) = TexifyIcons.LATEX_MODULE!!
}